pkg/cli/helm/helmclient.go

```diff
@@ -17,181 +17,99 @@ limitations under the License.
-
-		err := strvals.ParseIntoFile(arg, values, reader)
-		if err != nil {
-			return err
-		}
-	}
-
-	return nil
+func NewHelmClient() HelmClient {
+	return &HelmClientImpl{}
 }
 
-func runHelmInstall(helmConf *helm.Configuration, helmChart *chart.Chart, options ChartOptions) error {
+func (client *HelmClientImpl) RunHelmInstall(helmConf *helm.Configuration, helmChart *chart.Chart, releaseName, namespace string, wait bool) (*release.Release, error) {
 	installClient := helm.NewInstall(helmConf)
-	installClient.ReleaseName = options.ReleaseName
-	installClient.Namespace = options.Namespace
+	installClient.ReleaseName = releaseName
+	installClient.Namespace = namespace
 	installClient.CreateNamespace = true
-	installClient.Wait = true
 	installClient.Timeout = installTimeout
-	return runInstall(installClient, helmChart)
+	installClient.Wait = wait
+
+	return installClient.Run(helmChart, helmChart.Values)
 }
 
-func runHelmUpgrade(helmConf *helm.Configuration, releaseName string, helmChart *chart.Chart, options ChartOptions) error {
-	installClient := helm.NewUpgrade(helmConf)
-	installClient.Namespace = options.Namespace
-	installClient.Wait = true
-	installClient.Timeout = installTimeout
-	installClient.Recreate = true //force recreating radius pods on adding or modfying azprovider
-	return runUpgrade(installClient, options.ReleaseName, helmChart)
+func (client *HelmClientImpl) RunHelmUpgrade(helmConf *helm.Configuration, helmChart *chart.Chart, releaseName, namespace string, wait bool) (*release.Release, error) {
+	upgradeClient := helm.NewUpgrade(helmConf)
+	upgradeClient.Namespace = namespace
+	upgradeClient.Wait = wait
+	upgradeClient.Timeout = upgradeTimeout
+	upgradeClient.Recreate = true
+
+	return upgradeClient.Run(releaseName, helmChart, helmChart.Values)
 }
 
-// RunRadiusHelmUninstall attempts to uninstall Radius from the Radius system namespace
-// using a helm configuration, and returns an error if the uninstall fails.
-func RunHelmUninstall(helmConf *helm.Configuration, options ChartOptions) error {
-	output.LogInfo("Uninstalling %s from namespace: %s", options.ReleaseName, options.Namespace)
+func (client *HelmClientImpl) RunHelmUninstall(helmConf *helm.Configuration, releaseName, namespace string, wait bool) (*release.UninstallReleaseResponse, error) {
 	uninstallClient := helm.NewUninstall(helmConf)
 	uninstallClient.Timeout = uninstallTimeout
-	uninstallClient.Wait = true
-	_, err := uninstallClient.Run(options.ReleaseName)
-	if errors.Is(err, driver.ErrReleaseNotFound) {
-		output.LogInfo("%s not found", options.ReleaseName)
-		return nil
-	}
-	return err
+	uninstallClient.Wait = wait
+
+	return uninstallClient.Run(releaseName)
+}
+
+func (client *HelmClientImpl) RunHelmList(helmConf *helm.Configuration, releaseName, namespace string) ([]*release.Release, error) {
+	listClient := helm.NewList(helmConf)
+	listClient.Filter = releaseName
+	listClient.Deployed = true
+	listClient.AllNamespaces = true
+
+	return listClient.Run()
+}
```

Comment:
Should the namespace parameter be used? Maybe something like this?
```suggestion
func (client *HelmClientImpl) RunHelmList(helmConf *helm.Configuration, releaseName, namespace string) ([]*release.Release, error) {
    listClient := helm.NewList(helmConf)
    listClient.Filter = releaseName
    listClient.Deployed = true
    if namespace != "" {
        listClient.Namespace = namespace
        listClient.AllNamespaces = false
    } else {
        listClient.AllNamespaces = true
    }
    return listClient.Run()
}
```

Reply:
`listClient` doesn't accept a `.Namespace` parameter, so I'll remove it from the func definition

![image](https://github.com/user-attachments/assets/69ebe844-d11e-4998-8e29-9aa456b64570)